src/main/kotlin/lotto.kt

```diff
@@ -0,0 +1,38 @@
+class LottoGenerator {
+    fun generate(count: Int): List<List<Int>> {
+        val results = mutableListOf<List<Int>>()
+
+        repeat(count) {
+            val lotto_numbers = (1..45).shuffled().take(6).sorted()
+            results.add(lotto_numbers)
+        }
+
+        return results
+    }
+}
+
+fun main() {
+    var lottoCount: Int? = null
+
+    while (lottoCount == null || lottoCount !in 1..10) { // 1~10 범위로 제한
+        try {
+            print("구매하실 로또의 갯수를 입력해주세요 (최대 10개): ")
+            val input = readln().toIntOrNull() // 안전하게 문자열을 숫자로 변환
+            if (input != null && input in 1..10) {
+                lottoCount = input // 유효한 값일 경우 저장
+            } else {
+                println("1에서 10 사이의 숫자를 입력해주세요.")
+            }
+        } catch (e: Exception) {
+            println("유효하지 않은 입력입니다. 다시 시도해주세요.")
+        }
+    }
```

Comment:
JVM 계열 언에서는 최대한 null을 사용하지 않으려고합니다..! 
null 할당없이 처리하는 방식으로 코드를 짜보면 좋을 거 같습니다.

```suggestion
    var lottoCount = readLotttoCount();
    
fun readLotttoCount() {
  while(true) {
    val lottoCount = readin()....
    return lottoCount; // lottoCount 가 유효하다면 리턴
  }
} 
```